packages/fx-core/src/plugins/solution/fx-solution/v2/utils.ts

```diff
@@ -219,25 +223,34 @@ export async function checkWhetherLocalDebugM365TenantMatches(
     }
 
     if (maybeM365TenantId.value !== localDebugTenantId) {
-      const localFiles = [".fx/states/state.local.json"];
-
-      // add notification local file if exist
       if (
         projectPath !== undefined &&
         (await fs.pathExists(`${projectPath}/bot/.notification.localstore.json`))
       ) {
-        localFiles.push("bot/.notification.localstore.json");
+        const errorMessage = getLocalizedString(
+          "core.localDebug.tenantConfirmNotice",
+          localDebugTenantId,
+          maybeM365UserAccount.value,
+          "bot/.notification.localstore.json"
```

Comment:
Here should check the local state file `.fx/states/state.local.json` rather than the local storage file for notification app

Reply:
we will clear local envInfo here, so we won't fail here but just proceed.